yucca/preprocessing/YuccaPreprocessor.py

```diff
@@ -440,53 +441,58 @@ def reverse_preprocessing(self, images: np.ndarray, image_properties: dict):
         """
 
         nclasses = len(self.plans["dataset_properties"]["classes"])
-        original_shape = image_properties["original_shape"]
-        canvas = np.zeros((1, nclasses, *image_properties["uncropped_shape"]))
+        canvas = torch.zeros((1, nclasses, *image_properties["uncropped_shape"]), dtype=images.dtype)
         shape_after_crop = image_properties["cropped_shape"]
         shape_after_crop_transposed = shape_after_crop[self.transpose_forward]
         pad = image_properties["padding"]
 
-        for b in range(images.shape[0]):
-            for c in range(images.shape[1]):
-                image = images[b, c]
+        assert np.all(images.shape[2:] == image_properties["padded_shape"]), (
+            f"Reversing padding: "
+            f"image should be of shape: {image_properties['padded_shape']}"
+            f"but is: {images.shape[2:]}"
+        )
+        shape = images.shape[2:]
+        if len(pad) > 5:
+            images = images[
+                :,
+                :,
+                pad[0] : shape[0] - pad[1],
+                pad[2] : shape[1] - pad[3],
+                pad[4] : shape[2] - pad[5],
+            ]
+        elif len(pad) < 5:
```

Comment:
Doesn't this only work for `len(pad) == 4`? It would raise an `OutOfBounds error` if `len(pad) == 3`... I think we should change the if-statement to reflect this.